controller/model/posture_check_model_process_test.go

```diff
@@ -0,0 +1,195 @@
+)
+
+// Returns a process check and posture data that will pass with matching id, hash, signer, and running state. Can
+// be altered to test various pass/fail states
+func newMatchingProcessCheckAndData() (*PostureCheckProcess, *PostureData) {
+	postureCheckId := "30qhj45"
+	binaryHash := "b4f3228217a2bae3f21f6b6df3750d0723a5c3973db9aad360a8f25bc31e3676d38180cf0abc89d7fca7a26e1919a1e52739ed3116011acc7e96630313da56b8"
+	signerFingerprint := "950248b9e8b0dd41938018a871a13dd92bed4614"
+
+	postureResponse := &PostureResponse{
+		PostureCheckId: postureCheckId,
+		TypeId:         PostureCheckTypeProcess,
+		TimedOut:       false,
+		LastUpdatedAt:  time.Now(),
+	}
+
+	postureResponseProcess := &PostureResponseProcess{
+		IsRunning:         true,
+		BinaryHash:        binaryHash,
+		SignerFingerprint: signerFingerprint,
+	}
+
+	postureResponse.SubType = postureResponseProcess
+	postureResponseProcess.PostureResponse = postureResponse
+
+	validPostureData := &PostureData{
+		Mac:    nil,
+		Domain: nil,
+		Os:     nil,
+		Processes: []*PostureResponseProcess{
+			postureResponseProcess,
+		},
+	}
+
+	processCheck := &PostureCheckProcess{
+		PostureCheckId:  postureCheckId,
+		OperatingSystem: "Windows",
+		Path:            `C:\some\path\some.exe`,
+		Hashes: []string{
+			binaryHash,
+		},
+		Fingerprint: signerFingerprint,
+	}
+
+	return processCheck, validPostureData
+}
+
+func TestPostureCheckModelProcess_Evaluate(t *testing.T) {
+
+	t.Run("returns true for valid id, running, hash, and fingerprint", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+
+		result := processCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns false if not running", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		postureData.Processes[0].IsRunning = false
+
+		result := processCheck.Evaluate(postureData)
+		req := require.New(t)
+
+		req.False(result)
+	})
+
+	t.Run("returns true for valid id, running, hash, and fingerprint with mismatched hash case", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		postureData.Processes[0].BinaryHash = strings.ToUpper(postureData.Processes[0].BinaryHash)
+
+		result := processCheck.Evaluate(postureData)
+		req := require.New(t)
+
+		req.True(result)
+	})
+
+	t.Run("returns true for valid id, running, hash, and fingerprint with mismatched signer case", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		postureData.Processes[0].SignerFingerprint = strings.ToUpper(postureData.Processes[0].SignerFingerprint)
+
+		result := processCheck.Evaluate(postureData)
+		req := require.New(t)
+
+		req.True(result)
+	})
+
+	t.Run("returns true for valid id and running but check has null hashes and no signer", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		processCheck.Hashes = nil
+		processCheck.Fingerprint = ""
```

Comment:
what if the fingerprint gets leading/trailing whitespace? might be worth testing?

Reply:
I'll address in #400.